containers/src/Kokkos_DynRankView.hpp

```diff
@@ -1749,10 +1755,17 @@ KOKKOS_INLINE_FUNCTION constexpr auto DynRankView<D, P...>::layout() const ->
     case 6: return Impl::as_view_of_rank_n<6>(*this).layout();
     case 7: return Impl::as_view_of_rank_n<7>(*this).layout();
     default:
-      Kokkos::Impl::throw_runtime_exception(
-          "Calling DynRankView::layout on DRV of unexpected rank " +
-          std::to_string(rank()));
-  }
+      KOKKOS_IF_ON_HOST(
+          Kokkos::abort(
+              std::string(
+                  "Calling DynRankView::layout on DRV of unexpected rank " +
+                  std::to_string(rank()))
```

Comment:
There is no reason to wrap it in another `std::string` constructor call, right?
```suggestion
              ("Calling DynRankView::layout on DRV of unexpected rank " +
                  std::to_string(rank()))
```

Reply:
yes there is, I need to end up with a c-string

Reply:
and I'd rather have it explicit when writing `("some_string"+std::to_string()).c_str()` that that parenthesis creates a std::string